HSDS/Common/Application.h

```diff
@@ -163,6 +163,12 @@ class HSDS_Common_Export Application {
     DDS::ReturnCode_t retcode = DDS::RETCODE_OK;
     Unit<T>& unit = this->unit<T>();
 
+    const auto pos = unit.container.find(element);
+    if (pos != unit.container.end() && *pos == element) {
```

Comment:
Can the check `*pos == element` be included in `find` so that we don't have to do it every time `find` is called?

Reply:
I don't think I understand the question.  In this case, we are comparing the stored element to the new version from the user.  If they are equal, then there is no reason to insert and write which is the optimization that we wanted to achieve.

Reply:
What I meant was, should the comparison between the stored element and the new version happen inside of the new `find` method in `Container`? It could return a `list_.end()` if they are different and a valid `const_iterator` otherwise. The caller then just needs to check `if (pos != unit.container.end())`, similar to how it's used in `std`.

Reply:
Got it.  The `find` for the `Container` is more like a `map` then a `set`.  That is, you always care about the quality of the keys and you may care about the equality of the map value.  In the current implementation, `find` only cares about the keys which enables uses cased like retrieving a specific element.